Pivot/ViewController.swift

```diff
@@ -7,19 +7,135 @@
+            GLKMatrix3Multiply(GLKMatrix3MakeZRotation(-.pi / 2), embedVectorInMatrix(v2))
+        )
+        return angleDifference(v1, angleDirectionReferenceVector) - angleDifference(v2, angleDirectionReferenceVector)
+    }
+    
+    private func projectAttitudeOntoGravityPlane(_ rotationMatrix: GLKMatrix3,_ gravity: CMAcceleration,_ referenceVector: GLKVector3) -> GLKVector3 {
+        // understand the current attitude of the phone in terms of how it projects onto the plan defined by gravity as the normal to the plane (the x-y plane)
+        
+        // double check that gravityXRotation shouldn't use GLKMatrix3MakeXRotation
+        let gravityXRotation = GLKMatrix3MakeYRotation(Float(atan2(gravity.y, sqrt(pow(gravity.x,2) + pow(gravity.z,2)))))
+        let gravityYRotation = GLKMatrix3MakeYRotation(Float(atan2(gravity.x, gravity.z)))
 
+        // rotate the reference vector around the x and y axes to be parallel with gravity
+        var matrix = GLKMatrix3Multiply(rotationMatrix, gravityXRotation)
+        matrix = GLKMatrix3Multiply(matrix, gravityYRotation)
+        matrix = GLKMatrix3Multiply(matrix, embedVectorInMatrix(referenceVector))
+        
+        // throw away the z-coordinate, since we only care about the angle in the x-y plane
+        return GLKVector3Multiply(extractVectorFromMatrix(matrix), GLKVector3Make(1, 1, 0))
+    }
+    
+    private func embedVectorInMatrix(_ vector: GLKVector3) -> GLKMatrix3 {
+        let zeroVector = GLKVector3Make(0, 0, 0)
+        return GLKMatrix3MakeWithColumns( vector, zeroVector, zeroVector)
+    }
+    
+    private func extractVectorFromMatrix(_ matrix: GLKMatrix3) -> GLKVector3 {
+        return GLKMatrix3GetColumn(matrix, Int32(0))
+    }
+    
+    private func castCMRotationMatrixToGLKMatrix3(_ rM: CMRotationMatrix) -> GLKMatrix3 {
+        return GLKMatrix3Make(Float(rM.m11), Float(rM.m12), Float(rM.m13), Float(rM.m21), Float(rM.m22), Float(rM.m23), Float(rM.m31), Float(rM.m32), Float(rM.m33))
+    }
+    
+    private func castCMAccelerationToGLKVector3(_ acceleration: CMAcceleration) -> GLKVector3 {
+        return GLKVector3Make(Float(acceleration.x), Float(acceleration.y), Float(acceleration.z))
+    }
+    
+    // current UI is likely to change, don't worry about this method too much for now
+    private func updateUI() {
+        UIView.animate(withDuration: 0.1) {
+            self.pivotOrientationView.transform = CGAffineTransform(rotationAngle: CGFloat(self.total))
+        }
+        
+        let rotations = floor(abs(total / (2 * .pi)))
+        var direction = "left"
+        if rotations == 0 {
+            rotationsTextView.text = "Balanced"
+        } else {
+            if total > 0 {
+                direction = "right"
+            }
+            let rotationsString = String(rotations)
+            var rotationLabel = "rotation"
+            if rotations > 1 { rotationLabel = "rotations" }
```

Comment:
#?: Depending on the number of scenarios, this could use `switch` `case` for easy reading.